plugins/modem-manager/fu-mm-device.c

```diff
@@ -1291,6 +1343,22 @@ fu_mm_device_firehose_write(FuMmDevice *self,
 					 error);
 }
 
+static gboolean
+fu_mm_device_sahara_open(FuMmDevice *self, GError **error)
+{
+	self->sahara_loader = fu_sahara_loader_new(fu_device_get_physical_id(FU_DEVICE(self)));
```

Comment:
the ownership is a bit funky here -- is this really the best place to create the loader?